File: src/main/java/com/widehouse/whatnext/domain/Task.java
package com.widehouse.whatnext.domain;

import java.time.ZonedDateTime;
import java.util.Date;
import java.util.Objects;
import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String description;

    private Integer priority;

    private TaskStatus status;

    @ManyToOne
    private Category category;

    @CreatedDate
    private ZonedDateTime createdDate;

    public Task(String description, Integer priority, TaskStatus status, Category category) {
        this.description = description;
        this.priority = priority;
        this.status = status;
        this.category = category;
    }
}

File: src/main/kotlin/com/widehouse/whatnext/domain/Task.kt
package com.widehouse.whatnext.domain

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.ZonedDateTime
import javax.persistence.*

@Entity
@EntityListeners(AuditingEntityListener::class)
data class Task (
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int = 0,
        var description: String = "",
        var priority: Int = 1,
        var status: TaskStatus = TaskStatus.TODO,
        @ManyToOne
        var category: Category = Category(1, "", ""),
        @CreatedDate
        var createdDate: ZonedDateTime = ZonedDateTime.now()
) {
    constructor(description: String, priority: Int, status: TaskStatus, category: Category) : this() {
        this.description = description
        this.priority = priority
        this.status = status
        this.category = category
    }

}